libmeepgeom/ring-ll.cpp

```diff
@@ -151,6 +151,9 @@ int main(int argc, char *argv[])
 
   master_printf("all harminv results match reference values\n");
 
+  // this seems to be necessary to prevent failures
+  all_wait();
```

Comment:
Normally all serial code (like in harminv) is "MPI friendly", because each MPI process is a separate process with its own address space.